File: VB/12_LinksAndActions/ExtractAndUpdateLink.vb
Imports Spire.Pdf
Imports Spire.Pdf.Annotations
Imports System.ComponentModel
Imports System.Text
Imports System.Threading.Tasks

Namespace ExtractAndUpdateLink
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PdfDocument
			Dim doc As New PdfDocument()

			' Load an existing PDF document from the specified file path
			doc.LoadFromFile("..\..\..\..\..\..\Data\LinkAnnotation.pdf")

			' Get the first page of the document
			Dim page As PdfPageBase = doc.Pages(0)

			' Get the collection of annotations on the page
            Dim annotations As PdfAnnotationCollection = page.Annotations

			' Check if there are any annotations
			If annotations.Count > 0 Then
				' Traverse through the PdfAnnotationCollection
				For Each pdfAnnotation As PdfAnnotation In annotations
					' Check if it is a PdfTextWebLinkAnnotationWidget
					If TypeOf pdfAnnotation Is PdfTextWebLinkAnnotationWidget Then

						' Get the link annotation
						Dim annotation As PdfTextWebLinkAnnotationWidget = TryCast(pdfAnnotation, PdfTextWebLinkAnnotationWidget)

						' Change the URL of the link
						annotation.Url = "http://www.e-iceblue.com/Introduce/pdf-for-net-introduce.html"

					End If
				Next pdfAnnotation
			End If

			' Specify the output file path
			Dim result As String = "ExtractAndUpdateLink_out.pdf"

			' Save the modified document to the output file
			doc.SaveToFile(result)

			' Close the document
			doc.Close()

			' Launch the Pdf file
			PDFDocumentViewer(result)
		End Sub
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/12_LinksAndActions/ExtractAndUpdateLink.cs
using Spire.Pdf;
using Spire.Pdf.Annotations;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ExtractAndUpdateLink
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a pdf document
            PdfDocument doc = new PdfDocument();

            //Load file from disk
            doc.LoadFromFile(@"..\..\..\..\..\..\Data\LinkAnnotation.pdf");

            //Get the first page
            PdfPageBase page = doc.Pages[0];

            //Get the annotation collection
            PdfAnnotationCollection annotations = page.Annotations;

            //Verify whether widgetCollection is not null or not
            if (annotations.Count > 0)
            {
                //traverse the PdfAnnotationCollection
                foreach (PdfAnnotation pdfAnnotation in annotations)
                {
                    //if it is PdfTextWebLinkAnnotationWidget
                    if (pdfAnnotation is PdfTextWebLinkAnnotationWidget)
                    {

                        //Get the link annotation
                        PdfTextWebLinkAnnotationWidget annotation = pdfAnnotation as PdfTextWebLinkAnnotationWidget;

                        //Change the url
                        annotation.Url = "http://www.e-iceblue.com/Introduce/pdf-for-net-introduce.html";

                    }
                }
            }
            String result = "ExtractAndUpdateLink_out.pdf";

            //Save the document
            doc.SaveToFile(result);

            //Launch the Pdf file
            PDFDocumentViewer(result);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}





